mapadroid/worker/WorkerMITM.py

```diff
@@ -198,72 +212,56 @@ def __update_injection_settings(self):
-                else:
-                    self.logger.warning("No mode specified to wait for - this should not even happen...")
-                    time.sleep(0.5)
+        # proto has previously been received, let's check the timestamp...
+        mode = self._mapping_manager.routemanager_get_mode(self._routemanager_name)
+        timestamp_of_proto: float = latest_proto_entry.get("timestamp", None)
+        self.logger.debug("Latest timestamp: {} vs. timestamp waited for: {} of proto {}",
+                          datetime.fromtimestamp(timestamp_of_proto), datetime.fromtimestamp(timestamp),
+                          proto_to_wait_for)
+        if timestamp_of_proto < timestamp:
+            self.logger.debug("latest timestamp of proto {} ({}) is older than {}", proto_to_wait_for,
+                              timestamp_of_proto, timestamp)
+            # TODO: timeout error instead of data_error_counter? Differentiate timeout vs missing data (the
+            # TODO: latter indicates too high speeds for example
+            return type_of_data_found, data_found
+
+        latest_proto_data: dict = latest_proto_entry.get("values", None)
+        if latest_proto_data is None:
+            return LatestReceivedType.UNDEFINED, data_found
+        latest_proto = latest_proto_data.get("payload")
+        if mode in ["mon_mitm", "iv_mitm"]:
+            self.logger.debug("Checking GMO for mons")
+            # Now check if there are wild mons...
+            # TODO: Should we check if there are any spawnpoints? Wild mons could not be present in forests etc...
+            amount_of_wild_mons: int = 0
+            for cell in latest_proto['cells']:
+                amount_of_wild_mons += len(cell['wild_pokemon'])
+            if amount_of_wild_mons > 0:
+                data_found = latest_proto
+                type_of_data_found = LatestReceivedType.GMO
+            else:
+                self.logger.debug("No wild mons in GMO")
+        elif mode in ["raids_mitm"]:
+            self.logger.debug("Checking GMO for forts")
+            amount_of_forts: int = 0
+            for cell in latest_proto['cells']:
+                amount_of_forts += len(cell['forts'])
+            if amount_of_forts > 0:
+                data_found = latest_proto
+                type_of_data_found = LatestReceivedType.GMO
             else:
-                self.logger.debug("latest timestamp of proto {} ({}) is older than {}", proto_to_wait_for,
-                                  latest_timestamp, timestamp)
-                # TODO: timeout error instead of data_error_counter? Differentiate timeout vs missing data (the
-                # TODO: latter indicates too high speeds for example
-                time.sleep(0.5)
-        return data_requested
+                self.logger.debug("No forts in GMO")
+        else:
+            self.logger.warning("No mode specified to wait for - this should not even happen...")
+            time.sleep(0.5)
```

Comment:
Does this sleep contribute to the possibility of getting data back? If not we should get rid of it